src/plugins/utils/performance-utils.ts

```diff
@@ -0,0 +1,2 @@
+export const defaultPerformanceIgnore = (entry: PerformanceEntry) =>
```

Comment:
issue: We need tests for this function.

- *"when entry has a non-http URL schema then entry is ignored"*
- *"when entry has an http URL schema then entry is not ignored"*
- *"when entry is not type PerformanceResourceTiming then entry is not ignored"*

Reply:
Added unit tests for 
* 'when entry has a non-http URL schema then resource entry is ignored'
* 'when entry has an http URL schema then resource entry is not ignored'
* 'when entry has an https URL schema then resource entry is not ignored'

Did not add unit test for:
* 'when entry is not type PerformanceResourceTiming then entry is not ignored'. PerformanceResourceTiming extends PerformanceEntry, so any PerformanceResourceTiming will be recognized as a PerformanceEntry and will not be ignored.

Reply:
> any PerformanceResourceTiming will be recognized as a PerformanceEntry and will not be ignored

We need to test the path when `entry.entryType !== 'resource'`, which is currently not covered. This differentiates PerformanceResourceTiming from other subclasses of PerformanceEntry.

At a higher level, I am proposing that we ignore **resources** which do not use the **http(s)** schema. By allowlisting http(s), we ensure that we omit browser extension resources from multiple browsers (e.g., `moz-extension`, `chrome-extension`, etc.), and other unknown resources which are not relevant to the application being monitored.

However, this allowlist must only apply to PerformanceResourceTiming entries. When this function is used elsewhere (e.g., PerformanceNavigationTiming), we must not apply the allowlist.